test/unit/puppet_fact_parser_test.rb

```diff
@@ -220,6 +220,42 @@ def setup
     assert_equal '192.168.0.4', parser.interfaces['eth2'][:ipaddress]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    @importer = PuppetFactParser.new({'operatingsystem' => 'AnyOS', 'architecture' => 'AnyArc', 'operatingsystemrelease' => '6'})
+    arch = importer.architecture
+    os = importer.operatingsystem
+    assert arch.operatingsystems.include?(os)
+    assert os.architectures.include?(arch)
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    @importer = PuppetFactParser.new({'operatingsystem' => 'AnyOS', 'architecture' => 'AnyArc', 'operatingsystemrelease' => '6'})
+    os = importer.operatingsystem
+    arch = importer.architecture
+    assert arch.operatingsystems.include?(os)
+    assert os.architectures.include?(arch)
+  end
+
+  test "should create connection between existing os and existing arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    FactoryGirl.create(:architecture,:name => 'AnyArc')
+    @importer = PuppetFactParser.new({'operatingsystem' => 'AnyOS', 'architecture' => 'AnyArc', 'operatingsystemrelease' => '6'})
+    os = importer.operatingsystem
+    arch = importer.architecture
+    assert arch.operatingsystems.include?(os)
+    assert os.architectures.include?(arch)
+  end
+
+  test "should create connection between new os and existing arch on facts parsing" do
+    FactoryGirl.create(:architecture,:name => 'AnyArc')
+    @importer = PuppetFactParser.new({'operatingsystem' => 'AnyOS', 'architecture' => 'AnyArc', 'operatingsystemrelease' => '6'})
+    os = importer.operatingsystem
+    arch = importer.architecture
+    assert arch.operatingsystems.include?(os)
+    assert os.architectures.include?(arch)
```

Comment:
Tests seem to be ripe for a DRY refactor, perhaps a setup block or a helper function? At the end of the day it's the same code over and over.